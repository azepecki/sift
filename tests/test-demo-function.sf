def str search_if_present(str str1, str regex) {
        str return_val = "";
	if (reg_test(str1, regex)) {
		return_val = str1;
	}
        return return_val;
}

def int main() {

	str regex = "(w|m)i*ld";
	str string1 = "The wiiiiiiiiiild wild cat lived in a mild climate.";
	str string2 = "My name is Anthony Gonsalves.";

	search_if_present(string1, regex) |> print(); 
	search_if_present(string2, regex) |> print();
	
	return 0;

}
