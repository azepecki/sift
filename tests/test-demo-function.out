The wiiiiiiiiiild wild cat lived in a mild climate.
apple
